### PR TITLE
fix: selected patches order

### PR DESCRIPTION
**Pull request by mddanishansari**
fixed: #1331

### CODE REVIEW

File: lib/ui/widgets/patcherView/patch_selector_card.dart [hunk 1]
```diff
@@ -58,7 +59,9 @@ class PatchSelectorCard extends StatelessWidget {
 
   String _getPatchesSelection() {
     String text = '';
-    for (final Patch p in locator<PatcherViewModel>().selectedPatches) {
+    for (final Patch p in locator<PatcherViewModel>()
+        .selectedPatches
```
Comment (oSumAtrIX) [inline lib/ui/widgets/patcherView/patch_selector_card.dart:63]: Why not sort when initializing `selectedPatches`?

Comment (Ushie) [reply lib/ui/widgets/patcherView/patch_selector_card.dart:63]: Why not sort when generating the json overall? that results in it being sorted everywhere, patch selector screen, patchcard, etc

Comment (mddanishansari) [reply lib/ui/widgets/patcherView/patch_selector_card.dart:63]: I tried that first, but it fails in this scenario.


https://github.com/ReVanced/revanced-manager/assets/11080523/c017d67d-7dec-467b-a663-3669e5e14e1b

BTW, I had written sort logic earlier in `loadLastSelectedPatches()` function of `PatcherViewModel` just above `notifyListeners()`. Is that the correct place?

File: lib/ui/widgets/patcherView/patch_selector_card.dart [hunk 2]
```diff
@@ -1,3 +1,4 @@
+import 'package:collection/collection.dart';
```
Comment (validcube) [inline lib/ui/widgets/patcherView/patch_selector_card.dart:1]:
```suggestion
```

I don't think this package is related to sorting, please remove them or mark this as resolve.

👋 Anyway, thank for your contribution!

File: lib/ui/widgets/patcherView/patch_selector_card.dart (hunk 1 above)
Comment (mddanishansari) [reply lib/ui/widgets/patcherView/patch_selector_card.dart]: > Why not sort when generating the json overall? that results in it being sorted everywhere, patch selector screen, patchcard, etc

List is being sorted only for UI purpose and nothing else. That's why I think sorting it here in the view is correct. What do you think?

Comment (Ushie) [reply lib/ui/widgets/patcherView/patch_selector_card.dart]: In that case, I think your placement fits best

**Review by oSumAtrIX — APPROVED** [review]

**Review by validcube — APPROVED** [review]
🥞 Approved, thanks for your contribution!

**Pull request merged**